ngraph/frontend/onnx_import/src/op/org.openvinotoolkit/group_norm.cpp

```diff
@@ -0,0 +1,203 @@
+//*****************************************************************************
+// Copyright 2020 Intel Corporation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//*****************************************************************************
+
+#include "onnx_import/op/org.openvinotoolkit/group_norm.hpp"
+#include "ngraph/builder/reduce_ops.hpp"
+#include "ngraph/builder/split.hpp"
+#include "ngraph/node.hpp"
+#include "onnx_import/core/node.hpp"
+#include "onnx_import/default_opset.hpp"
+#include "onnx_import/utils/common.hpp"
+
+namespace ngraph
+{
+    namespace onnx_import
+    {
+        namespace op
+        {
+            namespace detail
+            {
+                namespace
+                {
+                    // This function creates a shape to which we need to reshape the input
+                    // before normalization.
+                    // If data shape is [N, C, H, W], then this function returns [N, num_groups, C // num_groups, H, W]
+                    std::shared_ptr<ngraph::Node>
+                        create_group_norm_shape(const Output<ngraph::Node>& data, size_t num_groups)
+                    {
+                        const auto& pshape = data.get_partial_shape();
+                        size_t rank_size = pshape.rank().get_length();
+                        NGRAPH_CHECK(rank_size == 4, "Only 4-D input tensors supported");
```

Comment:
Why are you support only 4D tensors? It's not so hard to make it work for the larger ones.
You can use the solution from the `create_scale_bias_shape` function.